velox/exec/Driver.cpp

```diff
@@ -340,22 +340,47 @@ void Driver::enqueueInternal() {
 
 // Call an Oprator method. record silenced throws, but not a query
 // terminating throw. Annotate exceptions with Operator info.
-#define CALL_OPERATOR(call, operator, methodName)                       \
-  try {                                                                 \
-    threadNumVeloxThrow() = 0;                                          \
-    call;                                                               \
-    recordSilentThrows(*operator);                                      \
-  } catch (const VeloxException& e) {                                   \
-    throw;                                                              \
-  } catch (const std::exception& e) {                                   \
-    VELOX_FAIL(                                                         \
-        "Operator::{} failed for [operator: {}, plan node ID: {}]: {}", \
-        methodName,                                                     \
-        operator->operatorType(),                                       \
-        operator->planNodeId(),                                         \
-        e.what());                                                      \
+#define CALL_OPERATOR(call, operator, methodName)                             \
+  try {                                                                       \
+    threadNumVeloxThrow() = 0;                                                \
+    opCallStatus_.withWLock([&](OpCallStatus& ocs) {                          \
+      ocs.timeStartMs = getCurrentTimeMs();                                   \
+      ocs.op = operator->operatorType();                                      \
+      ocs.method = methodName;                                                \
+    });                                                                       \
+    call;                                                                     \
+    opCallStatus_.withWLock([&](OpCallStatus& ocs) { ocs.timeStartMs = 0; }); \
```

Comment:
Let's review this change carefully. Adding locks and copying strings between each operator call may hurt performance for ML use cases.

CC: @kevinwilfong

Reply:
@mbasmanova , @kevinwilfong , @Yuhta 
Compared to what operators are doing, copying one string and acquiring one lock (that would almost never be held by anyone else) per method call... Not sure if that will hurt performance.

Do we have benchmarks to test this?

Should we wrap this into a Velox config property of gflag?